pkg/cluster/majorversionupgrade.go

```diff
@@ -54,6 +58,58 @@ func (c *Cluster) isUpgradeAllowedForTeam(owningTeam string) bool {
 	return util.SliceContains(allowedTeams, owningTeam)
 }
 
+func (c *Cluster) PatchAnnotations(patchBytesAnnotation []byte) error {
+	resourceName := c.Name
+	_, err := c.KubeClient.Postgresqls(c.Namespace).Patch(context.Background(), resourceName, types.MergePatchType, patchBytesAnnotation, metav1.PatchOptions{})
+	if err != nil {
+		c.logger.Errorf("failed to patch annotations: %v", err)
+		return err
+	}
+	return nil
+}
+
+func (c *Cluster) annotatePostgresResource(isSuccess bool) error {
+	annotations := make(map[string]string)
+	currentTime := metav1.Now().Format("2006-01-02T15:04:05Z")
+	if isSuccess {
+		annotations["last-major-upgrade-success"] = currentTime
+	} else {
+		annotations["last-major-upgrade-failure"] = currentTime
+	}
+
+	patchAnnotation := map[string]map[string]map[string]string{
+		"metadata": {
+			"annotations": annotations,
+		},
+	}
+	patchBytes, err := json.Marshal(patchAnnotation)
+	if err != nil {
+		c.logger.Errorf("failed to marshal annotations: %v", err)
+		return err
+	}
+
+	err = c.PatchAnnotations(patchBytes)
+	if err != nil {
+		c.logger.Errorf("failed to patch annotations to Postgres resource: %v", err)
+		return err
+	}
+	return nil
```

Comment:
so whenever there was a fail the `last-major-upgrade-failure` annotation will be present in the manifest and not touched again until it fails another time. Do we want to have it like this, or should it be set to nil?